File: python/doc/resources/plot_oat.py
"""Example code plotting one day's worth of outside air time-series.

@author Stephen Dawson-Haggerty <stevedh@eecs.berkeley.edu>
"""

from smap.archiver.client import SmapClient
from smap.contrib import dtutil

from matplotlib import pyplot
from matplotlib import dates

# make a client
c = SmapClient("http://www.openbms.org/backend")

# start and end values are Unix timestamps
start = dtutil.dt2ts(dtutil.strptime_tz("3-1-2013", "%m-%d-%Y"))
end   = dtutil.dt2ts(dtutil.strptime_tz("3-2-2013", "%m-%d-%Y"))

# hard-code the UUIDs we want to download
oat = [
  "395005af-a42c-587f-9c46-860f3061ef0d", 
  "9f091650-3973-5abd-b154-cee055714e59", 
  "5d8f73d5-0596-5932-b92e-b80f030a3bf7", 
  "ec2b82c2-aa68-50ad-8710-12ee8ca63ca7", 
  "d64e8d73-f0e9-5927-bbeb-8d45ab927ca5"
]

# perform the download
data = c.data_uuid(oat, start, end)

# plot all the data
#  use the epoch2num to convert to pylab date formats
for d in data:
  pyplot.plot_date(dates.epoch2num(d[:, 0] / 1000), d[:, 1], '-',
                   tz='America/Los_Angeles')

pyplot.show()

File: R/examples/plot_oat.R
library(RSmap)
RSmap("http://new.openbms.org/backend")

start <- as.numeric(strptime("3-29-2013", "%m-%d-%Y"))*1000
end <- as.numeric(strptime("3-31-2013", "%m-%d-%Y"))*1000

oat <- list("395005af-a42c-587f-9c46-860f3061ef0d",
         "9f091650-3973-5abd-b154-cee055714e59",
         "5d8f73d5-0596-5932-b92e-b80f030a3bf7",
         "d64e8d73-f0e9-5927-bbeb-8d45ab927ca5")

data <- RSmap.data_uuid(oat, start, end)

# returns a vector containing the min and max of the data
getExtents <- function(d){
  ex <- lapply(d, function(el){
    c(min(el$value), max(el$value))
  })
  ex <- unlist(ex)
  c(min(ex), max(ex))
}

ylim <- getExtents(data)

# convert to UTC seconds for R plot
time_UTC_sec <- data[[1]]$time/1000

# choose some pretty colors
col <- topo.colors(10)

# set up the plot and draw the first series
plot(time_UTC_sec
   , data[[1]]$value
   , xaxt="n"
   , type="l"
   , col=col[1]
   , ylim=ylim
   , xlab="Datetime"
   , ylab="Outside air temperature [°F]")

# format the x-axis to be the local time
axis.POSIXct(side=1, as.POSIXct(time_UTC_sec, origin="1970-01-01"),  format="%m-%d-%y")

# plot the rest of the series
for (i in 2:length(data)){
  lines(data[[i]]$time/1000, data[[i]]$value, col=col[i])  
}
